gensim/downloader.py

```diff
@@ -0,0 +1,359 @@
+from __future__ import absolute_import
+import argparse
+import json
+import os
+import tarfile
+import logging
+import sys
+import errno
+import hashlib
+try:
+    import urllib.request as urllib
+except ImportError:
+    import urllib
+
+try:
+    from urllib.request import urlopen
+except ImportError:
+    from urllib2 import urlopen
+
+user_dir = os.path.expanduser('~')
+base_dir = os.path.join(user_dir, 'gensim-data')
+data_log_file_path = os.path.join(base_dir, 'data.json')
+logger = logging.getLogger('gensim.api')
+
+
+def get_data_list():
+    """Function getting the list of all datasets/models.
+
+    Returns:
+        list: returns a list of datasets/models avalible for installation.
+    """
+    url = "https://raw.githubusercontent.com/RaRe-Technologies/gensim-data/master/list_with_filename.json"
+    response = urlopen(url)
+    data = response.read().decode("utf-8")
+    data = json.loads(data)
+    data_names = []
+    corpora = data['corpus']
+    models = data['model']
+    for corpus in corpora:
+        data_names.append(corpus)
+    for model in models:
+        data_names.append(model)
+    return data_names
+
+
+def get_data_name(data_):
```

Comment:
It will be a good idea to hide functions that no needed to user (with underscores)